zio/detector/file.go

```diff
@@ -129,7 +129,7 @@ func OpenFromNamedReadCloser(zctx *resolver.Context, rc io.ReadCloser, path stri
 	r := GzipReader(rc)
 	var zr zbuf.Reader
 	if cfg.Format == "" || cfg.Format == "auto" {
-		zr, err = NewReader(r, zctx)
+		zr, err = NewReader(r, zctx, path, cfg)
```

Comment:
I'm confused why this is needed when you do the same check that this does on line 140?

Reply:
The reader of line 140 is not the same as the throwaway reader that is created inside NewReader (as per the PR description). 

That said, maybe the `jsonConfig()` call of line 140 should also be done inside NewReader (or NewReaderWithConfig) rather than one here and one there.

Reply:
I see.